### DISCUSSION TITLE
OData API Queries are not working anymore due to URL-Encode

### DISCUSSION THREAD

**Discussion by bmt-sys-nd** [grafana/grafana-infinity-datasource#463] [Specific APIs]
Hi, we have plugin version 0.6.0 - OData queries are working fine.
We tried an update to 1.2.0 - OData queries do not work anymore.

The reason why is because the necessary `$` sign will be encoded to `%24`. However, `%24` will not be recognized by the OData API and therefor an error message will be returned rather than the queried data.

Example query for OData looks like this:
`https://odata-api.someserver.local/api/version/QUERY_PATH?$select=MyColumn1,MyColumn2&$filter=MyColumn2%20like%20%22%2Avalue%2A%22`

Unfortunately I don't see any possibility to disable URL-Encode in the plugin and make sure the otherwise correct encoding will be within the query.

So, this can be seen as a feature request to either way add a checkbox for OData (in order to not encode `$` in query strings), or to add the possibility to disable the encoding completely and take responsibility yoursrelf.

Thanks

**Comment by bmt-sys-nd** [comment]
@yesoreyeram You might have a look at the [OData documentation](https://learn.microsoft.com/en-us/odata/concepts/url-components):

> 3. Query options
Query options are essentially standardized querystring parameters that can be passed to an OData service to run queries on a requested resource. They can help perform certain operations like select, filter, count, skip, order, search and format on resources. **All OData query options are prefixed by a `$` sign and they are case-insensitive.**

It's not especially mentioned, but the `$` must be kept unencoded. The problem here with your plugin is, it does encode the `$` to `%24`.

The feature request in here is, to either way add the option to not encode the query at all, or to define a set of characters that should kept unencoded, like an excemtion list.

I hope it's clear now?

**Reply by yesoreyeram** [reply]
Yea makes sense. But I am running tight with other priorities.
If you have time and this is urgent, feel free to send a PR.

**Comment by bmt-sys-nd** [comment]
By the way I just learned that not only `$` must be kept unencode, but also `,`, `(` and `)`. We'll try to understand your code and fix it and probably send this as Pull Request as requested.

**Reply by bmt-sys-nd** [reply]
We found out that it's obviously not your code that does the encoding, but some library from Grafana does. We're not sure whether or not you can actually define exceptions there. Though, older versions of your plugin either way did no encoding at all, or used own code as it's still working with version 0.6.0 for example.

We're not any longer investigating but look for work arounds. Maybe you're interested in having OData properly supported with your plugin, maybe not. But this is as far as we got. Thanks.

**Reply by yesoreyeram** [reply]
Sorry. I am no longer investigating this. Not a priority at the moment. You can continue using the old version if required.
